fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/SerializableRegistriesMixin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2023 The Quilt Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.registry.sync;
+
+import java.util.stream.Stream;
+
+import org.spongepowered.asm.mixin.Dynamic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.quiltmc.qsl.registry.api.sync.RegistrySynchronization;
+
+import net.minecraft.registry.SimpleRegistry;
+import net.minecraft.registry.DynamicRegistryManager;
+import net.minecraft.registry.SerializableRegistries;
+
+import net.fabricmc.fabric.impl.registry.sync.DynamicRegistriesImpl;
+
+// Implements skipping empty dynamic registries with the SKIP_WHEN_EMPTY sync option.
+@Mixin(SerializableRegistries.class)
+abstract class SerializableRegistriesMixin {
+	@Shadow
+	private static Stream<DynamicRegistryManager.Entry<?>> stream(DynamicRegistryManager dynamicRegistryManager) {
+		return null;
+	}
+
+	@Dynamic("method_45961: Codec.xmap in createDynamicRegistryManagerCodec")
+	@Redirect(method = "method_45961", at = @At(value = "INVOKE", target = "Lnet/minecraft/registry/SerializableRegistries;stream(Lnet/minecraft/registry/DynamicRegistryManager;)Ljava/util/stream/Stream;"))
+	private static Stream<DynamicRegistryManager.Entry<?>> filterNonSyncedEntries(DynamicRegistryManager drm) {
+		return stream(drm).filter(entry -> {
+			//boolean canSkip = DynamicRegistriesImpl.SKIP_EMPTY_SYNC_REGISTRIES.contains(entry.key());
+			//return !canSkip || entry.value().size() > 0;
```

Comment:
maybe a comment explaining this? the bridge seems very solid though!

Reply:
I'm honestly not sure if it even works as a proper bridge, mainly because the Registry Sync API is experimental and undocumented—mainly went on a hunch when writing this... going to ping patbox in the main thread for this pr about this since GitHub is jank and doesn't let you ping inside subthreads

Reply:
Does this work as a bridge for FAPI's version of optional Dynamic Registries or will it need a fix on QSL's side to properly support? @Patbox

The full relevant code is at https://github.com/cocona20xx/quilted-fabric-api/blob/0.86.0-upstream-compat/fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/SerializableRegistriesMixin.java, since commit view is a bit annoying to actually get a look at code in.

I *think* it'll work, if I'm understanding QSL's `RegistrySynchronization#isEntryPresent` correctly, but I think it's best to actually ask you since you're the author of said API and probably know best here.

Reply:
Quilt's registry sync was always intended to be only used for static registries. It also uses flag system for entries. So if entry is optional, isEntryPresent will return whatever true if client has that optional entry, otherwise not

Reply:
Will be improved in an upcoming commit thanks to QSL changes.